active/0000-stateful-helpers.md

```diff
@@ -0,0 +1,322 @@
+- Start Date: 2015-05-17
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+Stateful helpers are a class-based way to define helpers. Stateful helpers:
+
+  * Have a single return value
+  * Can store and read state
+  * Have lifecycle hooks analagous to components where appropriate. For
+    example, a stateful helper may call `recompute` at any time to generate a new
+    value.
+  * Are a superset of traditional function-based helpers. They can do more,
+    but in many cases a traditional helper is appropriate.
+
+They fill a gap in Ember's current template APIs:
+
+|                  | has positional params | has layout | can yield template | has lifecycle, instance | can control rerender |
```

Comment:
I don't necessarily feel like this table is adding a lot to understanding the problem, and it makes it look like each of the columns is a legitimate "primitive knob" that you might want to tune separately. I also think the comparison with Component makes it seem even more like there are knobs being tuned.

The way I'd talk about it is:

* Components are stateful and manage an area of DOM
* Helpers produce a single value that can be passed into other helpers, but don't directly control an area of DOM
  * `Ember.Helper`s have a full lifecycle and access to services, and have the ability to trigger a recomputation
    when something external changes.
  * There is a shorthand for `Ember.Helper` that you can use if you have no dependencies on external services.